src/Nethermind/Nethermind.State/Proofs/TxTrie.cs

```diff
@@ -38,8 +36,8 @@ public TxTrie(IReadOnlyList<Transaction>? txs, bool allowMerkleProofConstruction
             // a temporary trie would be a trie that exists to create a state root only and then be disposed of
             for (int i = 0; i < txs.Count; i++)
             {
-                Rlp transactionRlp = _txDecoder.Encode(txs[i], RlpBehaviors.SkipTypedWrapping);
-                Set(Rlp.Encode(i).Bytes, transactionRlp.Bytes);
+                Span<byte> transactionRlp = MixedEncoding.Encode(txs[i], RlpBehaviors.SkipTypedWrapping);
```

Comment:
Looks like a consensus bug. We should change decoding, but starting from CancunTimestamp

Reply:
MixedEncoding does not change old transactions encoding, and if no blob transaction involved, it will work the same

Reply:
(If it changes it's a bug indeed)

Reply:
got it, thx